extended/src/main/java/apoc/vectordb/VectorEmbeddingHandler.java

```diff
@@ -0,0 +1,190 @@
+package apoc.vectordb;
+
+import apoc.ml.RestAPIConfig;
+import org.neo4j.internal.kernel.api.procs.ProcedureCallContext;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import static apoc.ml.RestAPIConfig.BODY_KEY;
+import static apoc.ml.RestAPIConfig.JSON_PATH_KEY;
+import static apoc.ml.RestAPIConfig.METHOD_KEY;
+import static apoc.util.MapUtil.map;
+import static apoc.vectordb.VectorEmbeddingConfig.*;
+
+public interface VectorEmbeddingHandler {
+
+    <T> VectorEmbeddingConfig fromGet(Map<String, Object> config,
+                                      ProcedureCallContext procedureCallContext,
+                                      List<T> ids);
+
+    VectorEmbeddingConfig fromQuery(Map<String, Object> config,
+                                    ProcedureCallContext procedureCallContext,
+                                    List<Double> vector,
+                                    Object filter,
+                                    long limit,
+                                    String collection);
+    
+    //
+    // -- implementations
+    //
+    
+    class QdrantEmbeddingHandler implements VectorEmbeddingHandler {
```

Comment:
I wonder if we should move these implementations closer to where the vector databases are? either into the procedures file or an associated file? Otherwise we have to update this file whenever we add a new db?

Reply:
Moved